src/lib/cache.coffee

```diff
@@ -66,6 +66,18 @@ module.exports = (dnschain) ->
                     @cache.setex(key, ttl, JSON.stringify value) if not err2
                     valueDoer err, key, value
 
+        resolveResource: (resolver, resource, property, operation, fmt, args, cb) ->
+            if @blockchainEnabled and resolver.cacheTTL?
+                retriever = (key, callback) =>
+                    f = (err, result) =>
+                        callback err, resolver.cacheTTL, result
+                    resolver.resources[resource].call resolver, property, operation, fmt, args, f
+                doer = (err, key, result) =>
+                    cb err, result
+                @get "#{resolver.name}:#{resource}:#{property}:#{operation}:#{fmt}:#{JSON.stringify(args)}", retriever, doer
+            else
+                resolver.resources[resource].call resolver, property, operation, fmt, args, cb
+
```

Comment:
Are you certain this new function needs to be added? Is there any way that the existing `resolveBlockchain` can be used?